proxy_manager.go

```diff
@@ -29,7 +29,7 @@ import (
 
 type proxyManager struct {
 	mu              *sync.RWMutex
```

Comment:
Can't we remove this mutex now? 🤔

Reply:
Yes, good catch, removed at: https://github.com/hazelcast/hazelcast-go-client/pull/922/commits/75741c9ffe1fb3a94937b48de519458410fdf894